app/src/lib/stores/app-store.ts

```diff
@@ -4933,9 +4937,25 @@ export class AppStore extends TypedBaseStore<IAppState> {
           lfsRepositories.push(refreshedRepo)
         }
       } else {
-        const error = new Error(`${path} isn't a git repository.`)
-        this.emitError(error)
+        invalidPaths.push(path)
+      }
+    }
+
+    if (invalidPaths.length > 0) {
+      let errorMessage = `The following paths aren't git repositories:\n\n${invalidPaths
+        .slice(0, MaxInvalidFoldersToDisplay)
+        .map(path => `- ${path}`)
+        .join('\n')}${
+        invalidPaths.length > MaxInvalidFoldersToDisplay
+          ? `\n\n(and ${invalidPaths.length - MaxInvalidFoldersToDisplay} more)`
+          : ''
+      }`
+
+      if (invalidPaths.length === 1) {
+        errorMessage = `${invalidPaths} isn't a git repository.`
```

Comment:
Nitpicking deluxe here but could we refactor this so that we only go down the map/join path if there are multiple invalid paths? Additionally we should capitalize `Git` in `git repositories`

```suggestion
      let errorMessage

      if (invalidPaths.length === 1) {
        errorMessage = `${invalidPaths} isn't a Git repository.`
      } else {
        errorMessage = `The following paths aren't Git repositories:\n\n${invalidPaths
          .slice(0, MaxInvalidFoldersToDisplay)
          .map(path => `- ${path}`)
          .join('\n')}${
          invalidPaths.length > MaxInvalidFoldersToDisplay
            ? `\n\n(and ${
                invalidPaths.length - MaxInvalidFoldersToDisplay
              } more)`
            : ''
        }`
```

This could be make much nicer with a little helper method for producing  the error message and using early returns

Reply:
Thanks for the suggestion! I'm going to create a separate method since I agree it's going to be much nicer!